src/main/java/org/folio/rest/impl/BLUsersAPI.java

```diff
@@ -770,7 +787,12 @@ public void getBlUsersSelf(List<String> include, boolean expandPerms,
           Context vertxContext) {
     String token = okapiHeaders.get(OKAPI_TOKEN_HEADER);
     String username = getUsername(token);
-    run(null, username, expandPerms, include, okapiHeaders, asyncResultHandler);
+    String userId = getUserId(token);
+    if (StringUtils.isBlank(username) || username.startsWith(UNDEFINED_USER) || StringUtils.isBlank(userId)) {
+      run(null, username, expandPerms, include, okapiHeaders, asyncResultHandler);
+    } else {
+      run(userId, null, expandPerms, include, okapiHeaders, asyncResultHandler);
+    }
```

Comment:
Can you please explain what this new `if` check is doing here?

Reply:
@steveellis we put this info in description

Reply:
Thank you for clarifying this in the description.